### PR TITLE
feat:view port features queue

### PR DESCRIPTION
**Pull request by padawannn**

### CODE REVIEW

File: src/viz/interactivity/viewport-features/ViewportFeaturesQueue.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+export class ViewportFeaturesQueue {
```
Comment (VictorVelarde) [inline src/viz/interactivity/viewport-features/ViewportFeaturesQueue.ts:1]: This looks like an ingenious and practical approach. Cool! ✨

File: src/viz/layer/Layer.ts [hunk 2]
```diff
@@ -510,14 +514,23 @@ export class Layer extends WithEvents implements StyledLayer {
...
+        if (this.filtersCollection.hasFilters()) {
+          features = features.filter(feature => filters.applicator(feature));
+        }
+
+        this._viewportFeaturesQueue.resolveQueue(features);
+      });
     }
 
-    return features;
+    return (await pendingPromise.catch(error => console.debug(error))) || [];
```
Comment (VictorVelarde) [inline src/viz/layer/Layer.ts:533]: I'm wondering if we could hide a bit the complexity from the queue, under the ViewportFeaturesGenerator itself....

I'm thinking about instantiating _viewportFeaturesQueue inside it, and let the public method getFeatures make the IF-EMPTY-XXXX internally.... what do you think?

**Review by VictorVelarde — APPROVED** [review]
Great! Just a general comment.

**Comment by VictorVelarde** [comment]
Let's wait just to have first #130  and we're ready to go

File: src/viz/layer/Layer.ts (hunk 2 above)
Comment (padawannn) [reply src/viz/layer/Layer.ts:533]: I like this approach but we would need to refactor some things. I prefer to keep it as it is for the moment

**Pull request merged**